### PR TITLE
feat: wishlist management

### PR DESCRIPTION
**Pull request by PuneetP16**
## VIDEO PREVIEW

https://user-images.githubusercontent.com/91944618/160257641-8591ffde-bc98-42a7-b5a9-7c2bc3257989.mp4
## Pages to Check
WishlistContext
HorizontalCard
ProductCard
CartContext
useAxios
CartReducer

### CODE REVIEW

File: src/customHooks/useAxios.js [hunk 1]
```diff
@@ -2,12 +2,12 @@ import axios from "axios";
...
 
 export const useAxios = () => {
-	const { isAuth, token } = useAuth();
+	const { token } = useAuth();
 	const { toggleLoader } = useLoader();
 	let headers = {};
 	let output;
 
-	const axiosRequest = async ({ method, url, resKey, data = {} }) => {
+	const axiosRequest = async ({ method, url, resKey, data = {}}) => {
```
Comment (tarunsankhla) [inline src/customHooks/useAxios.js]: no need t0 pass data as object it is empty object by default

File: src/pages/Wishlist/Wishlist.jsx [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
 import "./Wishlist.css";
 
 export const Wishlist = () => {
 	useDocumentTitle("Wishlist | MS");
+	const { axiosRequest } = useAxios();
+	const { wishlist, totalQty, wishlistDispatch, getUpdatedWishlist } =
+		useWishlist();
+
+	// useEffect(() => {
+	// 	(async () => {
```
Comment (tarunsankhla) [inline src/pages/Wishlist/Wishlist.jsx:12]: remove unwanted comment

File: src/pages/Wishlist/Wishlist.jsx [hunk 3]
```diff
@@ -1,39 +1,52 @@
...
+import { Link } from "react-router-dom";
 import { Listing } from "../../components";
-import { useDocumentTitle } from "../../customHooks";
+import { useWishlist } from "../../contexts";
+import { useAxios, useDocumentTitle } from "../../customHooks";
 import "./Wishlist.css";
 
 export const Wishlist = () => {
 	useDocumentTitle("Wishlist | MS");
+	const { axiosRequest } = useAxios();
```
Comment (tarunsankhla) [inline src/pages/Wishlist/Wishlist.jsx]: remove useaxios if not in use

File: src/reducers/cartReducer.js [hunk 4]
```diff
@@ -33,22 +36,25 @@ export const cartReducer = (
 			return { cart, totalQty, totalPrice };
 
 		case "UPDATE":
+			console.log("update");
 			cart = [...payload];
 			// Client side update
 			// cart = cart.filter((p) => p._id !== payload._id);
```
Comment (tarunsankhla) [inline src/reducers/cartReducer.js:36]: remove unwanted and commented code

File: src/reducers/cartReducer.js [hunk 5]
```diff
@@ -4,7 +4,10 @@ export const cartReducer = (
 ) => {
 	switch (type) {
 		case "ADD":
+			
+
 			cart = [...payload];
```
Comment (tarunsankhla) [inline src/reducers/cartReducer.js]: abstract out the spread of payload on top of switch condition

**Review by tarunsankhla — APPROVED** [review]
Overall it's great, I learned a few things about Axios called a custom hook, just minor changes, and good to merge

File: src/customHooks/useAxios.js (hunk 1 above)
Comment (PuneetP16) [reply src/customHooks/useAxios.js]: I too thought so and I tried it but it was giving undefined not object. So I tried by myself by creating a small example.
Here's the link : [Must have Defaults](https://onecompiler.com/javascript/3xxa97hb4)

File: src/pages/Wishlist/Wishlist.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/pages/Wishlist/Wishlist.jsx]: done.

File: src/reducers/cartReducer.js (hunk 5 above)
Comment (PuneetP16) [reply src/reducers/cartReducer.js]: Done

File: src/reducers/cartReducer.js (hunk 4 above)
Comment (PuneetP16) [reply src/reducers/cartReducer.js:36]: Done

File: src/pages/Wishlist/Wishlist.jsx (hunk 2 above)
Comment (PuneetP16) [reply src/pages/Wishlist/Wishlist.jsx:12]: I have a plan of implementing it so kept.

**Review by m26unkwn — COMMENTED** [review]
overall Code is good and it is readable.

**Review by m26unkwn — APPROVED** [review]
overall Code is good and it is readable.
1. There is a problem with the loader while upgrading the item.

**Comment by PuneetP16** [comment]
> overall Code is good and it is readable.

Will check it. Thanks for review

**Pull request merged**